src/braket/circuits/free_parameter.py

```diff
@@ -0,0 +1,76 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+from sympy import Symbol
+
+from braket.circuits.free_parameter_expression import FreeParameterExpression
+
+
+class FreeParameter(FreeParameterExpression):
+    """
+    Class 'FreeParameter'
+    """
+
+    def __init__(self, name: str):
+        """
+        Initializes a new :class:'FreeParameter' object.
+        Free parameters can be used in parameterized circuits.
+
+        Args:
+            name (str): Name of the :class:'FreeParameter'. Can be a unicode value.
+
+        Examples:
+            >>> param1 = FreeParameter("theta")
+            >>> param1 = FreeParameter("\u03B8")
+        """
+        self._name = Symbol(name)
+        super().__init__(expression=self._name)
+
+    @property
+    def name(self) -> str:
+        """
+        str: Name of this parameter.
+        """
+        return self._name.name
+
+    def subs(self, parameter_values):
+        """
+        Substitutes a value in if the parameter exists within the mapping.
+
+        Args:
+            parameter_values: A mapping of parameter to its corresponding value.
+
+        Returns: The substituted value otherwise returns the object.
```

Comment:
Needs type hints. Also, the `Returns` docstring should read something like
> The substituted value if this parameter is in parameter_values, otherwise returns self